doc/doxygen/src/debugging-aids.md

```diff
@@ -0,0 +1,36 @@
+# Debugging Tools {#debugging-tools}
+
+## Undefined Behavior Sanitizer (ubsan) {#ubsan}
+
+RIOT contains makefile convenience support for gcc/clang's undefined
+behaviour sanitizer.
+
+### Overview
+
+Both gcc and clang allow generation on code that does runtime checks for
+undefined behavior (UB).
+
+E.g., the following code might trigger UB for some parameters:
+
+```C
+void test(int foo) {
+    return (foo << 24);
+}
+```
+
+In this case, the signed shift would be alright unless:
+
+- it would "push out" all bits to the left, with undefined runtime result. Here,
+  that happens on architectures with 16-bit integers.
+- `foo` is negative, with implementation defined runtime results.
+
+Using ubsan, these can be caught at runtime.
+
+### How to use
+
+1. build with `make all-ubsan`. This will cause a trap on undefined
+   behaviour.
+
+2. build with `UBSAN_MODE=msg_exit make all-ubsan` will print where and how
+   the undefined behaviour occured, then exit with error code. This is only
+   available on native built with gcc.
```

Comment:
Wouldn't setting msg_exit be a good default for native? (Just tested, as usual first without looking at all docs, and was surprised to see a trap rather than an error message).

Reply:
totally. I've updated both the .mk and the docs.